setuptools/tests/test_egg_info.py

```diff
@@ -79,7 +79,8 @@ def run():
     @staticmethod
     def _extract_mv_version(pkg_info_lines: List[str]) -> Tuple[int, int]:
         version_str = pkg_info_lines[0].split(' ')[1]
-        return tuple(map(int, version_str.split('.')[:2]))
+        major, minor, *_ = map(int, version_str.split('.'))
```

Comment:
Isn't this doing more computing?

(Previously we were restricting the `map` to the first 2 elements, now we are running it over everything)

I don't know, but there might be a chance the rest of the string may raise an exception when converted to `int`, isn't it?

Reply:
Hmm, looking back at it, you're right. I think I can get the best of both worlds to avoid
mypy: `error: Incompatible return value type (got "Tuple[int, ...]", expected "Tuple[int, int]")`
pyright:
```
Expression of type "tuple[int, ...]" cannot be assigned to return type "Tuple[int, int]"
  "tuple[int, ...]" is incompatible with "Tuple[int, int]"
    Tuple size mismatch; expected 2 but received indeterminate
```
by doing
```py
>>> version_str = "1.2.345-beta"
>>> major, minor = map(int, version_str.split('.')[:2])
>>> major, minor
(1, 2)
```